rust/connlib/libs/client/src/messages.rs

```diff
@@ -190,7 +191,8 @@ mod test {
                 "interface": {
                     "ipv4": "100.72.112.111",
                     "ipv6": "fd00:2021:1111::13:efb9",
-                    "upstream_dns": []
+                    "upstream_dns": [],
+                    "dns_fallback_strategy": "upstream_resolver"
```

Comment:
@jamilbk I think adding this to the protocol should be made in another PR where we also add handling for it.

Reply:
Hmm, yeah I see that it's a ball of yarn now. I'll revert this dns stuff.

Reply:
Ah, you've done that. Thanks!